@balloob, FYI not sure what to do here with this check. It failed on:

`homeassistant/helpers/device_registry.py:45:4: R1705: Unnecessary "elif" after "return" (no-else-return)`

But that is not something I changed. Travis also ran successful on my branch.

--- a/homeassistant/components/media_player/directv.py
+++ b/homeassistant/components/media_player/directv.py
@@ -241,7 +241,7 @@ class DirecTvDevice(MediaPlayerDevice):
         if self._is_standby:
             return None
 
-       return self._last_position
+        return self._last_position
 
     @property
     def media_position_updated_at(self):
@@ -269,23 +269,6 @@ class DirecTvDevice(MediaPlayerDevice):
             return None
 
         return self._current.get('episodeTitle')
-
-    @property
-    def media_channel(self):
-        """Return the channel current playing media."""
-        if self._is_standby:
-            return None
-
-        return "{} ({})".format(
-            self._current['callsign'], self._current['major'])
-
-    @property
-    def source(self):
-        """Name of the current input source."""
-        if self._is_standby:
-            return None
-
-        return self._current['major']
 
     @property
     def media_channel(self):

--- a/homeassistant/components/media_player/directv.py
+++ b/homeassistant/components/media_player/directv.py
@@ -130,7 +130,7 @@ class DirecTvDevice(MediaPlayerDevice):
         self._paused = None
         self._last_position = None
         self._is_recorded = None
-        self._is_client = False if device == '0' else True
+        self._is_client = device != '0'
         self._assumed_state = None
         self._available = False
 
@@ -336,15 +336,19 @@ class DirecTvDevice(MediaPlayerDevice):
 
     def turn_on(self):
         """Turn on the receiver."""
-        if not self._is_client:
-            _LOGGER.debug("Turn on %s", self._name)
-            self.dtv.key_press('poweron')
+        if self._is_client:
+            raise NotImplementedError()
+
+        _LOGGER.debug("Turn on %s", self._name)
+        self.dtv.key_press('poweron')
 
     def turn_off(self):
         """Turn off the receiver."""
-        if not self._is_client:
-            _LOGGER.debug("Turn off %s", self._name)
-            self.dtv.key_press('poweroff')
+        if self._is_client:
+            raise NotImplementedError()
+
+        _LOGGER.debug("Turn off %s", self._name)
+        self.dtv.key_press('poweroff')
 
     def media_play(self):
         """Send play command."""

--- a/homeassistant/components/media_player/directv.py
+++ b/homeassistant/components/media_player/directv.py
@@ -36,6 +36,10 @@ SUPPORT_DTV = SUPPORT_PAUSE | SUPPORT_TURN_ON | SUPPORT_TURN_OFF | \
     SUPPORT_PLAY_MEDIA | SUPPORT_SELECT_SOURCE | SUPPORT_STOP | \
     SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK | SUPPORT_PLAY
 
+SUPPORT_DTV_CLIENT = SUPPORT_PAUSE | \
+    SUPPORT_PLAY_MEDIA | SUPPORT_SELECT_SOURCE | SUPPORT_STOP | \
+    SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK | SUPPORT_PLAY
+
 DATA_DIRECTV = 'data_directv'
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
@@ -126,10 +130,15 @@ class DirecTvDevice(MediaPlayerDevice):
         self._paused = None
         self._last_position = None
         self._is_recorded = None
+        self._is_client = False if device == '0' else True
         self._assumed_state = None
         self._available = False
 
-        _LOGGER.debug("Created DirecTV device for %s", self._name)
+        if self._is_client:
+            _LOGGER.debug("Created DirecTV client %s for device %s",
+                          self._name, device)
+        else:
+            _LOGGER.debug("Created DirecTV device for %s", self._name)
 
     def update(self):
         """Retrieve latest state."""
@@ -290,7 +299,7 @@ class DirecTvDevice(MediaPlayerDevice):
     @property
     def supported_features(self):
         """Flag media player features that are supported."""
-        return SUPPORT_DTV
+        return SUPPORT_DTV_CLIENT if self._is_client else SUPPORT_DTV
 
     @property
     def media_currently_recording(self):
@@ -327,13 +336,15 @@ class DirecTvDevice(MediaPlayerDevice):
 
     def turn_on(self):
         """Turn on the receiver."""
-        _LOGGER.debug("Turn on %s", self._name)
-        self.dtv.key_press('poweron')
+        if not self._is_client:
+            _LOGGER.debug("Turn on %s", self._name)
+            self.dtv.key_press('poweron')
 
     def turn_off(self):
         """Turn off the receiver."""
-        _LOGGER.debug("Turn off %s", self._name)
-        self.dtv.key_press('poweroff')
+        if not self._is_client:
+            _LOGGER.debug("Turn off %s", self._name)
+            self.dtv.key_press('poweroff')
 
     def media_play(self):
         """Send play command."""

